python/lsst/daf/base/propertyContainer/propertyContainerContinued.py

```diff
@@ -700,6 +700,35 @@ def items(self):
     def values(self):
         return ValuesView(self)
 
+    def pop(self, name, default=None):
+        """Remove the named key and return its value.
+
+        Parameters
+        ----------
+        name : `str`
+            Name of the key to remove.
+        default : Any, optional
+            Value to return if the key is not present.
+
+        Returns
+        -------
+        value : Any
+            The value of the item as would be returned using `getScalar()`.
+
+        Raises
+        ------
+        KeyError
+            Raised if no default is given and the key is missing.
+        """
+        if name in self:
+            value = self[name]
+            self.remove(name)
+        else:
+            if default is None:
+                raise KeyError(f"{name} not found.")
```

Comment:
I believe the convention (or at least "what `dict` does") is to make missing key the only argument passed to `KeyError`.